src/geometry/euclidean.lean

```diff
@@ -683,4 +683,230 @@ begin
+    { rw [←vsub_right_mem_direction_iff_mem (self_mem_mk' _ _) _, direction_mk',
+          vsub_vadd_eq_vsub_sub, vadd_vsub_assoc, add_comm, add_sub_assoc],
+      refine submodule.add_mem _ (orthogonal_projection_fn_vsub_mem_direction_orthogonal hn hc p) _,
+      rw submodule.mem_orthogonal',
+      intros w hw,
+      rw [←neg_sub, inner_neg_left, orthogonal_projection_inner_eq_zero hc _ w hw, neg_zero] },
+    have hm : (orthogonal_projection hc) v +ᵥ orthogonal_projection_fn hn hc p ∈
+      ({orthogonal_projection_fn hn hc (v +ᵥ p)} : set P),
+    { rw ←inter_eq_singleton_orthogonal_projection_fn hn hc (v +ᵥ p),
+      exact set.mem_inter hs ho },
+    rw set.mem_singleton_iff at hm,
+    exact hm.symm
+  end }
+
+/-- The intersection of the subspace and the orthogonal subspace
+through the given point is the `orthogonal_projection` of that point
+onto the subspace. -/
+lemma inter_eq_singleton_orthogonal_projection {s : affine_subspace ℝ V P}
+    (hn : (s : set P).nonempty) (hc : is_complete (s.direction : set V)) (p : P) :
+  (s : set P) ∩ (mk' p s.direction.orthogonal) = {orthogonal_projection hn hc p} :=
+inter_eq_singleton_orthogonal_projection_fn hn hc p
+
+/-- The `orthogonal_projection` lies in the given subspace. -/
+lemma orthogonal_projection_mem {s : affine_subspace ℝ V P} (hn : (s : set P).nonempty)
+    (hc : is_complete (s.direction : set V)) (p : P) : orthogonal_projection hn hc p ∈ s :=
+orthogonal_projection_fn_mem hn hc p
+
+/-- The `orthogonal_projection` lies in the orthogonal subspace. -/
+lemma orthogonal_projection_mem_orthogonal {s : affine_subspace ℝ V P}
+    (hn : (s : set P).nonempty) (hc : is_complete (s.direction : set V)) (p : P) :
+  orthogonal_projection hn hc p ∈ mk' p s.direction.orthogonal :=
+orthogonal_projection_fn_mem_orthogonal hn hc p
+
+/-- Subtracting a point in the given subspace from the
+`orthogonal_projection` produces a result in the direction of the
+given subspace. -/
+lemma orthogonal_projection_vsub_mem_direction {s : affine_subspace ℝ V P}
+    (hn : (s : set P).nonempty) (hc : is_complete (s.direction : set V)) {p1 : P} (p2 : P)
+    (hp1 : p1 ∈ s) :
+  orthogonal_projection hn hc p2 -ᵥ p1 ∈ s.direction :=
+vsub_mem_direction (orthogonal_projection_mem hn hc p2) hp1
+
+/-- Subtracting the `orthogonal_projection` from a point in the given
+subspace produces a result in the direction of the given subspace. -/
+lemma vsub_orthogonal_projection_mem_direction {s : affine_subspace ℝ V P}
+    (hn : (s : set P).nonempty) (hc : is_complete (s.direction : set V)) {p1 : P} (p2 : P)
+    (hp1 : p1 ∈ s) :
+  p1 -ᵥ orthogonal_projection hn hc p2 ∈ s.direction :=
+vsub_mem_direction hp1 (orthogonal_projection_mem hn hc p2)
+
+/-- A point equals its orthogonal projection if and only if it lies in
+the subspace. -/
+lemma orthogonal_projection_eq_self_iff {s : affine_subspace ℝ V P}
+    (hn : (s : set P).nonempty) (hc : is_complete (s.direction : set V)) (p : P) :
```

Comment:
We can make `p` implicit because it occurs on both sides of the iff.
```suggestion
    (hn : (s : set P).nonempty) (hc : is_complete (s.direction : set V)) {p : P} :
```